tests/integration/models/test_team.py

```diff
@@ -0,0 +1,154 @@
+import asyncio
+import uuid
+
+import pytest
+
+from synapseclient import Synapse
+from synapseclient.core.exceptions import SynapseHTTPError
+from synapseclient.models import Team
+from synapseclient.models.user import UserGroupHeader
+
+
+class TestTeam:
+    """Integration tests for Team."""
+
+    @pytest.fixture(autouse=True, scope="function")
+    def init(self, syn: Synapse) -> None:
+        self.syn = syn
+        self.expected_name = "test_team_" + str(uuid.uuid4())
+        self.expected_description = "test description"
+        self.expected_icon = None
+        self.team = Team(
+            name=self.expected_name,
+            description=self.expected_description,
+            icon=self.expected_icon,
+        )
+        self.TEST_USER = "test_account_synapse_client"
+        self.TEST_MESSAGE = "test message"
+
+    @pytest.mark.asyncio
+    async def test_create(self) -> None:
+        # GIVEN a team object self.team
+        # WHEN I create the team on Synapse
+        test_team = await self.team.create()
+        # THEN I expect the created team to be returned
+        assert test_team.id is not None
+        assert test_team.name == self.expected_name
+        assert test_team.description == self.expected_description
+        assert test_team.icon is None
+        assert test_team.etag is not None
+        assert test_team.created_on is not None
+        assert test_team.modified_on is not None
+        assert test_team.created_by is not None
+        assert test_team.modified_by is not None
+        # Clean up
+        await test_team.delete()
+
+    @pytest.mark.asyncio
+    async def test_delete(self) -> None:
+        # GIVEN a team object self.team
+        # WHEN I create the team on Synapse
+        test_team = await self.team.create()
+        # AND I delete the team
+        await test_team.delete()
+        # THEN I expect the team to no longer exist
+        with pytest.raises(SynapseHTTPError):
```

Comment:
nit: We could verify the return message. For example I had done this in Activity:
```
        with pytest.raises(
            client.SynapseHTTPError, match="404 Client Error: \nNo activity"
        ):
```